ui/src/app/views/workspace-edit/component.tsx

```diff
@@ -0,0 +1,590 @@
+        return;
+      }
+      this.setState({workspace : this.props.workspace});
+      if (this.isMode(WorkspaceEditMode.Clone)) {
+        this.setState(
+          fp.set(['workspace', 'name'], ('Duplicate of ' + this.props.workspace.name)));
+      }
+    }
+
+    async setCdrVersions() {
+      const cdrVersions = await cdrVersionsApi().getCdrVersions();
+      // Convert cdrVersion to select input format of label and value
+      const versions = [];
+      cdrVersions.items.map((version , i) => {
+        versions.push({label: version.name, value: version.cdrVersionId});
+      });
+      this.setState({cdrVersionItems: versions});
+      if (this.isMode(WorkspaceEditMode.Create)) {
+        this.setState(fp.set(['workspace', 'cdrVersionId'], versions[0].value));
+      }
+    }
+
+
+    openStigmatization() {
+      window.open('/definitions/stigmatization', '_blank');
+    }
+
+    renderHeader() {
+      switch (this.props.routeConfigData.mode) {
+        case WorkspaceEditMode.Create:
+          return 'Create a new Workspace';
+        case WorkspaceEditMode.Edit:
+          return 'Edit workspace \"' + this.props.workspace.name + '\"';
+        case WorkspaceEditMode.Clone:
+          return 'Clone workspace \"' +
+              this.props.workspace.name + '\"';
+      }
+    }
+
+    renderButtonText() {
+      switch (this.props.routeConfigData.mode) {
+        case WorkspaceEditMode.Create: return 'Create Workspace';
+        case WorkspaceEditMode.Edit: return 'Update Worspace';
+        case WorkspaceEditMode.Clone: return 'Duplicate Workspace';
+      }
+    }
+
+    disableButton() {
+      return this.isEmpty('name') || this.isEmpty('description');
+    }
+
+    updateWorkspaceCategory(category, value) {
+      this.setState(fp.set(['workspace', 'researchPurpose', category], value));
+    }
+
+
+    updateUnderserverPopulation(populationDetails) {
+      this.setState(fp.set(['workspace', 'researchPurpose', 'underservedPopulationDetails'],
+        populationDetails));
+      this.setState(fp.set(['workspace', 'researchPurpose', 'containsUnderservedPopulation'],
+        (populationDetails.length > 0)));
+    }
+
+    async saveWorkspace() {
+      try {
+        this.setState({loading: true});
+        let workspace = this.state.workspace;
+        if (this.isMode(WorkspaceEditMode.Create)) {
+          workspace =
+              await workspacesApi().createWorkspace(this.state.workspace);
+        } else if (this.isMode(WorkspaceEditMode.Clone)) {
+          await workspacesApi().cloneWorkspace(
+            this.state.workspace.namespace, this.state.workspace.id,
+            {
+              includeUserRoles: this.state.cloneUserRole,
+              workspace: this.state.workspace
+            });
+        } else {
+          await workspacesApi()
+              .updateWorkspace(this.state.workspace.namespace, this.state.workspace.id,
+                  {workspace: this.state.workspace});
```

Comment:
When I changed one of the checkboxes on this, and updated workspace, I got a null pointer exception on the backend. It looks like it may not be passing up research purpose?

Reply:
I found this on a clone as well, but only the second clone I did?

Reply:
I am able to clone and update workspace. But there is some inconsistency regarding etag, i am checking that

Reply:
Thank you for checking!